### PR TITLE
drm/vc4: Allow option to transpose the output on the writeback connector

### PR DESCRIPTION
**Pull request by 6by9**
The hardware behind the writeback connector can do a transpose as it writes the image out.
DRM offers no mechanism to select that, but as a simple test we can note when the width/height of the buffer are swapped wrt the mode, and enable the transpose. The atomic_check obviously has to be updated too to permit that combination.

Transposing a square output buffer is left as future problem to solve.

### CODE REVIEW

**Comment by 6by9** [comment]
@jc-kynesim PR updated. It also includes all the other pending DRM tweaks that are kicking around for you/DT.

**Comment by 6by9** [comment]
JC has said he's happy with this, so drop the draft tag.

The checkpatch complaint on spaces before tabs is deliberate as it matches all the other comments in that block.

Branch rebased.

File: drivers/gpu/drm/drm_connector.c [hunk 1]
```diff
@@ -362,7 +362,7 @@ static int __drm_connector_init(struct drm_device *dev,
 	drm_object_attach_property(&connector->base,
 				   config->non_desktop_property,
 				   (connector_type != DRM_MODE_CONNECTOR_VIRTUAL &&
-				   connector_type != DRM_MODE_CONNECTOR_WRITEBACK) ? 0 : 1;
+				   connector_type != DRM_MODE_CONNECTOR_WRITEBACK) ? 0 : 1);
```
Comment (popcornmix) [inline drivers/gpu/drm/drm_connector.c:365]: This should be squashed in previous commit.

Comment (6by9) [reply drivers/gpu/drm/drm_connector.c:365]: Done.

**Pull request merged**